### DISCUSSION TITLE
Worse results with newest fast dreambooth?

### DISCUSSION THREAD

**Discussion by yvz38** [TheLastBen/fast-stable-diffusion#1930] [General]
I recently had to update to the newest colab version of fast dreambooth and noticed a big loss in likeness of the person I've trained. I don't know if I'm hallucinating, so to be sure, I trained a person again, who I had trained a while ago already (with an older version of the colab and with the person's permission ofc).

I've used the exact same settings in both, same lr, steps, same images, same filename (token), everything was the same. It was really just the newest colab version. I triple checked. Compared them both and the newest version's results look like some totally different person.
At first I thought it was because I was using a different model as the base, but then I switched back to the default sd 1.5 base and got the same worse results.

Here is a comparison of the old vs. new with the same seed:

Old version:

![grid-1108](https://user-images.githubusercontent.com/22767528/229386538-7d0f2262-37ab-4237-af34-fc69040b9c92.jpg)
---
New version:

![grid-1109](https://user-images.githubusercontent.com/22767528/229386559-65bb5813-7148-4dcc-bae1-a31feb015122.jpg)

For reference, here is a real photo of the person that I took:

![IMG_6153-1 400x400](https://user-images.githubusercontent.com/22767528/229387016-3ba6d0e4-0322-4287-8873-e67b2e165445.jpg)

I'm really stuck and don't know what to do, which is why I'm posting this here. I'd like to do more comparisons but I think I've reached my free colab for today.

**Comment by TheLastBen** [comment]
Are you 100% sure that the base model use in both results is the same ?

**Reply by meunumerotim333** [reply]
Something change on update Ben, maybe even you haven't investigate enought. I don't know technical details, but I do know old versions still way better than new ones. I'm not complaining, just warning, because maybe it needs to revert updates to increase the quality results for users. 

Or I have an idea, you could create a new tab on your github for old DreamBooth for alternative and old school results:

![image](https://user-images.githubusercontent.com/121369033/229409060-ed0b7ee0-6828-49a4-8094-00738a3f6ba9.png)

A suggest this one: https://colab.research.google.com/drive/1uUUGu5bH1u2RML-XYm01E9ME3KngpZmW#scrollTo=R3SsbIlxw66N

**Reply by yvz38** [reply]
Thanks for your reply and yes I'm sure that on both of those I had the used the same SD 1.5 base. I chose 1.5 in the colab UI and left the text input fields empty for custom model.

I tried to do another before & after compare, but I'm still having issues with free colab. Today I was running out of VRAM (CUDA out of memory) for some reason. And I've noticed that training the Text Encoder took way longer than before or fails directly. But I assume that's an issue on my side/bad luck? Google colab may assigned me a GPU with low VRAM, cuz they didn't have better ones available today.

I will try again tomorrow, because I want be sure as well.

And yes, I agree that a working old version would be amazing. The colab version that I copied on 10.03.2023 served me very well. It would be perfect if there were a way to fix just the error " No module named 'wget' " when trying to dl the model in the 3rd cell.

**Comment by wmgp2021** [comment]
Its not your imagination. SD on Colab has been dogshit lately. Its an uphill battle for Ben to keep it running vs Google ruining it again.
The results iv been getting the last 2 weeks have been terrible. Very blurry, unclear, many artifacts and alot of crashes.
Using the exact same checkpoints and prompts. Its just giving terrible results period.

**Reply by meunumerotim333** [reply]
I'm using since december getting the same great results as I'm used to because I'm using the same Colab, I'm just updating the depencies when Ben updates. So I don't think the problem is on Goggle Colab.. something is changing on the code between updates.

Try to use this one: https://colab.research.google.com/drive/1uUUGu5bH1u2RML-XYm01E9ME3KngpZmW#scrollTo=R3SsbIlxw66N

Is the one I'm using and is working pretty well. I've never tested new updates again after was giving me bad results.

**Reply by KodyAG** [reply]
> I'm using since december getting the same great results as I'm used to because I'm using the same Colab, I'm just updating the depencies when Ben updates. So I don't think the problem is on Goggle Colab.. something is changing on the code between updates.
> 
> Try to use this one: https://colab.research.google.com/drive/1uUUGu5bH1u2RML-XYm01E9ME3KngpZmW#scrollTo=R3SsbIlxw66N
> 
> Is the one I'm using and is working pretty well. I've never tested new updates again after was giving me bad results.

I've used this and the resemblance is very good, but it's very hard to change the style... I've used 200 class images for 15 photos, should I be using more or what?

**Reply by meunumerotim333** [reply]
> > I'm using since december getting the same great results as I'm used to because I'm using the same Colab, I'm just updating the depencies when Ben updates. So I don't think the problem is on Goggle Colab.. something is changing on the code between updates.
> > Try to use this one: https://colab.research.google.com/drive/1uUUGu5bH1u2RML-XYm01E9ME3KngpZmW#scrollTo=R3SsbIlxw66N
> > Is the one I'm using and is working pretty well. I've never tested new updates again after was giving me bad results.
> 
> I've used this and the resemblance is very good, but it's very hard to change the style... I've used 200 class images for 15 photos, should I be using more or what?

I only train close faces.. its my style. I normally use 15 pictures x 100 steps (1500 steps)

This colab here is also old but have more options, if you want to use need to replace for new dependecies and model download:
https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb
